lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -341,6 +342,30 @@ protected function getCoverImageAlt() {
   }
 
   /**
+  * Get status whether this campaign has a competition or not.
+  *
+  * @return bool
+  */
+  protected function getCompetition($id) {
+    $query = db_select('node', 'n');
+    $query->leftJoin('dosomething_signup_data_form', 'sdf', 'n.nid = sdf.nid');
+    $query->fields('sdf', ['competition_signup']);
+    $query->condition('n.nid', $id);
+
+    $result = $query->execute()->fetchAll();
+    $result = $result[0]->competition_signup;
+
+    if ($result == '1') {
+      $result = TRUE;
+    }
+    elseif ($result == '0') {
+      $result = FALSE;
+    }
+
+    return $result;
+  }
```

Comment:
For queries that need to be built out, could we move it off into a `dosomething_campaigns.query.inc` file or whatever? similar to how we do with Reportbacks, ReportbackItems, and Signups?